docs/backend.md

```diff
@@ -0,0 +1,138 @@
+
+### Data migrations
+
+We should have a principled way of approaching data schema migrations. They **will** happen, some known examples being:
+
+- introducing some grouping for features within a scheme
+- trimming coordinate precision
+- evolving a v2 schema for describing interventions in more detail
+
+In an ideal situation, we can express migrations as two lossless functions, that transform the old to the new, and the new to the old. This will not be (completely) true for the examples above. So, we have to decide, possibly on a case-by-case basis, who is responsible for updating the data, if it can't be done automatically. For example, after we settle on a v2 schema, will we ask some or all LAs to fill out new forms for existing interventions? Will we attempt to automate this for them, using some kind of text parsing on the freeform description? What if the new schema requires them to split a linear route every time the cross-section design changes?
+
+## API
+
+The data model and storage are likely the difficult parts of this design; the API and server implementing it should be much simpler. They can just be a thin **stateless** wrapper, performing authentication and then interacting with the database. Thus, they could be serverless, since the start-up time should be trivial.
+
+We should check if GDS prefers any particular languages or frameworks. Some options that some/all of the team is familiar with:
+
+- Go
+- Python (Flask?) -- is the GIL a problem?
+- Rust (overkill, Dustin not in favor)
+- Others?
+
+We can figure out the JSON (or gRPC?) API as we go. The output would generally be an ATIP GeoJSON file, or the currently-evolving combined scheme GeoJSON format. Some endpoints would be:
+
+- get one scheme + version
+- get the latest version of a scheme
+- update one scheme
+- get all schemes combined (possibly filtering the results)
+
+### Svelte / UI changes
+
+We need to integrate whatever GDS auth layer. Talking to the backend may be trivial (`fetch` calls), or maybe we use something like websockets for saving commands while people edit.
+
+We should be very careful to keep the existing ATIP pages (browsing authorities, editing a scheme, and browsing a bunch of combined schemes) able to be used **without** any of the backend and database this doc describes. The current file-based approach is very simple to deploy. There may be non-ATE users who don't want any sort of backend. So, we need to find a way to make the frontend changes agnostic to a backend (the current one being "local storage").
```

Comment:
Exactly, I think simple config should be fine. Right now there's a handful of places doing things with `window.localStorage`; we'd refactor some kind of interface to do those things, implement it twice, and use a bit of config to instantiate the right one. Complexity will grow as we make components for login, sharing, undo/redo, etc, but I don't think any of it will be too hard. Just flagging that as soon as we start working on the backend, we should do this from the start, and not be caught in a difficult refactor later